src/Analyzers/CSharp/Analyzers/SimplifyLinqExpressions/CSharpSimplifyLinqExpressionsDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,158 @@
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+
+        protected override void InitializeWorker(AnalysisContext context)
+        {
+            context.RegisterCompilationStartAction(OnCompilationStart);
+        }
+
+        private void OnCompilationStart(CompilationStartAnalysisContext context)
+        {
+            var whereMethods = ImmutableHashSet.CreateBuilder<IMethodSymbol>();
+            var linqMethods = ImmutableHashSet.CreateBuilder<IMethodSymbol>();
+
+            var enumNamedType = context.Compilation.GetTypeByMetadataName(typeof(System.Linq.Enumerable).FullName);
+            if (enumNamedType is null)
+            {
+                return;
+            }
+
+            var enumMethods = enumNamedType.GetMembers(nameof(Enumerable.Where)).OfType<IMethodSymbol>();
+
+            if (enumMethods is null)
+            {
+                return;
+            }
+
+            AddIfNotNull(whereMethods, enumMethods);
+
+            // add all valid linq calls
+            foreach (var id in s_validLinqCalls)
+            {
+                enumMethods = enumNamedType?.GetMembers(id).OfType<IMethodSymbol>();
+                AddIfNotNull(linqMethods, enumMethods);
+            }
+
+            if (whereMethods.Count > 0 && linqMethods.Count > 0)
+            {
+                context.RegisterOperationAction(
+                    context => AnalyzeAction(
+                        context, whereMethods.ToImmutable(), linqMethods.ToImmutable()), OperationKind.Invocation);
+            }
+
+            return;
+
+            static void AddIfNotNull(ImmutableHashSet<IMethodSymbol>.Builder set, IEnumerable<IMethodSymbol>? others)
+            {
+                if (others != null)
+                {
+                    set.UnionWith(others);
+                }
+            }
+        }
+
+        public void AnalyzeAction(OperationAnalysisContext context, ImmutableHashSet<IMethodSymbol> whereMethods, ImmutableHashSet<IMethodSymbol> linqMethods)
+        {
+            var invocationOperation = (IInvocationOperation)context.Operation;
+            var semanticModel = context.Operation.SemanticModel;
+
+            var firstChildArgument = invocationOperation.Children.FirstOrDefault(c => c is IArgumentOperation);
+
+            if (firstChildArgument is null)
+            {
+                return;
+            }
+
+            var whereClause = firstChildArgument.Children.FirstOrDefault(c => c is IInvocationOperation);
+
+            if (whereClause is null)
+            {
+                return;
+            }
+
+            var isLambda = whereClause.Children.Any(c => c.Syntax is ArgumentSyntax argument && argument.Expression is SimpleLambdaExpressionSyntax);
```

Comment:
```suggestion
            var isLambda = whereClause.Children.Any(c => c.Syntax is { ArgumentSyntax: { Expression: SimpleLambdaExpressionSyntax } });
```

Also, this won't work for: `.Where((s) => s.Length > 0)` or `.Where((string s) => s.Length > 0);`